tools/xcodeproj_shims/output-processor.rb

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env ruby
+
+BAZEL_WORKSPACE = ENV['BAZEL_WORKSPACE_ROOT'].freeze
+
+# Documentation
+class BazelOutputLine
+  attr_reader :text
+
+  def initialize(line)
+    @text = line
+  end
+
+  # Try to create a processed line based on a match rule, or a pass-through
+  def self.from(line)
+    if (message_line = MessageLine.create_from(line))
+      message_line
+    elsif (snapshot_line = SnapshotLine.create_from(line))
+      snapshot_line
+    else
+      BazelOutputLine.new(line)
+    end
+  end
+end
+
+# Documentation
+class RegexMatchLine < BazelOutputLine
+  attr_reader :regex, :match_data
+
+  @regex = nil
+  @match_data = nil
+
+  def self.create_from(line)
+    rule = new(line)
+
+    rule.match_data.nil? ? nil : rule
+  end
+end
+
+# Documentation
+class SnapshotLine < RegexMatchLine
+  def initialize(line)
+    @regex = /^DEBUG: .+ Snapshot test (.+) (\(.+\)) does not specify it uses an app host$/
```

Comment:
this is square-specific, this i think should just match _all_ lines that begin with `DEBUG` and are followed by a path to a starlark file